src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,72 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
+
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instences = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("The class " + clazz.getName()
+                    + " does not have the @Component annotation");
+        }
+        Object clazzImplementationInstance = createNewInstance(clazz);
+
+        Field[] declaredFiles = clazz.getDeclaredFields();
+        for (Field field: declaredFiles) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                try {
+                    field.setAccessible(true);
+                    field.set(clazzImplementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can`t initialize field value. "
+                            + "Class:" + clazz.getName() + "Field: " + field.getName());
+                }
+            }
+        }
+        return clazzImplementationInstance;
+    }
+
+    private Object createNewInstance(Class<?> clazz) {
+        if (instences.containsKey(clazz)) {
+            return instences.get(clazz);
+        }
+        try {
+            Constructor<?> constructor = clazz.getConstructor();
+            Object instance = constructor.newInstance();
+            instences.put(clazz, instance);
+            return instance;
+        } catch (ReflectiveOperationException e) {
```

Comment:
It is better to replace many exceptions that have a common parent with a general parental exception. In this case, 'ReflectiveOperationException' can be used instead of specifying each exception individually.